src/checkLinuxWebAppDownDetector.ts

```diff
@@ -0,0 +1,175 @@
+    facility = "Facility",
+    failureCount = "FailureCount"
+}
+
+export const detectorCancelTokens: Map<string, CancellationTokenSource> = new Map();
+export async function checkLinuxWebAppDownDetector(node: SiteTreeItem, tokenSource: CancellationTokenSource): Promise<void> {
+    return await callWithTelemetryAndErrorHandling('appService.linuxWebAppDownDetector', async (context: IActionContext): Promise<void> => {
+        const deployment: DeployResult = await node.root.client.kudu.deployment.getResult('latest');
+        if (!deployment.startTime) {
+            // if there's no deployment detected, nothing can be done
+            context.telemetry.properties.cancelStep = 'noDeployResult';
+            return;
+        }
+
+        const deployResultTime: Date = new Date(deployment.startTime);
+
+        const detectorOutput: string = `Diagnosing web app "${node.root.client.siteName}" for critical errors...`;
+        ext.outputChannel.appendLine(detectorOutput);
+
+        const detectorUri: string = `${node.root.environment.resourceManagerEndpointUrl}${node.id}/detectors/${detectorId}`;
+        const requestOptions: WebResource & Partial<{ qs: queryString }> = new WebResource();
+
+        requestOptions.method = 'GET';
+        requestOptions.url = detectorUri;
+        requestOptions.qs = {
+            'api-version': "2015-08-01",
+            fId: "1",
+            btnId: "2",
+            inpId: "1",
+            val: "vscode",
+            startTime: deployResultTime.toISOString(),
+            endTime: new Date().toISOString()
+        };
+        await signRequest(requestOptions, node.root.credentials);
+
+        let detectorErrorMessage: string | undefined;
+        // wait 10 minutes for a response from the detector
+        const detectorTimeoutMs: number = Date.now() + 20 * 60 * 1000;
+        do {
+            if (tokenSource.token.isCancellationRequested) {
+                // the user cancelled the check by deploying again
+                throw new UserCancelledError();
+            }
+
+            if (Date.now() > detectorTimeoutMs) {
+                const noIssuesFound: string = `No critical issues found for web app "${node.root.client.siteName}".`;
+                ext.outputChannel.appendLine(noIssuesFound);
+                return undefined;
+            }
+            // update the new end time to be now
+            requestOptions.qs.endTime = new Date().toISOString();
+            detectorErrorMessage = await getLinuxDetectorError(requestOptions, deployResultTime, node.root.client.fullName);
+            // poll every minute
+            await delay(1000 * 60);
+        } while (!detectorErrorMessage);
+
+        await ext.ui.showWarningMessage(detectorErrorMessage, { title: 'Open in Portal' });
+        const portalDeeplink = `${node.root.environment.portalUrl}/?websitesextension_ext=asd.featurePath%3Ddetectors%2F${detectorId}#@microsoft.onmicrosoft.com/resource/${node.root.client.id}/troubleshoot`;
```

Comment:
Please use `openInPortal` from the UI package

Reply:
Unrelated, but in this PR, you have a comment regarding unconventional portal URLs and I'd like to pull that same card 😓

Reply:
This is what openInPortal does:
```
export async function openInPortal(root: types.ISubscriptionContext, id: string, options?: types.OpenInPortalOptions): Promise<void> {
    const queryPrefix: string = (options && options.queryPrefix) ? `?${options.queryPrefix}` : '';
    const url: string = `${root.environment.portalUrl}/${queryPrefix}#@${root.tenantId}/resource${id}`;

    await openUrl(url);
}
```

Can't you pass `websitesextension_ext=asd.featurePath%3Ddetectors%2F${detectorId}` as the query and `${node.root.client.id}/troubleshoot` as the id?

My url looks like this:
```
${treeItem.root.environment.portalUrl}/#blade/AppInsightsExtension/QuickPulseBladeV2/ComponentId/${componentId}/ResourceId/${resourceId}
```

It doesn't have `#@microsoft.onmicrosoft.com/resource` like yours does. You have a bug because you're not using tenantId (you always use "microsoft.onmicrosoft.com" as the tenant). That's why I wanted you to use the shared one